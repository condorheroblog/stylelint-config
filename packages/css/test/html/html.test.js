import { readFileSync } from "node:fs";
import { join } from "node:path";
import stylelint from "stylelint";
import { beforeEach, describe, expect, test } from "vitest";

import basicConfig from "../../";

const config = { ...basicConfig, ignoreFiles: [] };
const validHtmlPath = join(__dirname, "./valid.html");
const invalidHtmlPath = join(__dirname, "./invalid.html");
const validHtml = readFileSync(validHtmlPath, "utf-8");
const invalidHtml = readFileSync(invalidHtmlPath, "utf-8");

describe("flags no warnings in HTML with valid css", () => {
  let result;

  beforeEach(() => {
    result = stylelint.lint({
      config,
      files: validHtmlPath,
    });
  });

  test("did not error", () => {
    return result.then(data => expect(data.errored).toBeFalsy());
  });

  test("flags no warnings", () => {
    return result.then(data => expect(data.results[0].warnings).toHaveLength(0));
  });
});

describe("flags warnings in HTML with invalid css", () => {
  let result;

  beforeEach(() => {
    result = stylelint.lint({
      config,
      files: invalidHtmlPath,
    });
  });

  test("did error", () => {
    return result.then(data => expect(data.errored).toBeTruthy());
  });

  test("flags one warning", async() => {
    return result.then(data => expect(data.results[0].warnings).toHaveLength(11));
  });

  test("correct warning text", () => {
    return result.then(data =>
      expect(data.results[0].warnings[0].text).toBe(
        "Expected \"width\" to come before \"color\" (order/properties-order)",
      ),
    );
  });

  test("correct rule flagged", () => {
    return result.then(data =>
      expect(data.results[0].warnings[0].rule).toBe("order/properties-order"),
    );
  });

  test("correct severity flagged", () => {
    return result.then(data => expect(data.results[0].warnings[0].severity).toBe("error"));
  });

  test("correct line number", () => {
    return result.then(data => expect(data.results[0].warnings[0].line).toBe(9));
  });

  test("correct column number", () => {
    return result.then(data => expect(data.results[0].warnings[0].column).toBe(37));
  });
});

describe("HTML pieces to be tested", () => {
  test("valid.html", () => {
    return expect(validHtml).toMatchSnapshot();
  });

  test("invalid.html", () => {
    return expect(invalidHtml).toMatchSnapshot();
  });
});
