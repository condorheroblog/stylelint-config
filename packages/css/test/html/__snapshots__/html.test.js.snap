// Vitest Snapshot v1

exports[`HTML pieces to be tested > invalid.html 1`] = `
"<!DOCTYPE html>
<html lang=\\"en\\">
<head>
    <meta charset=\\"UTF-8\\">
    <meta http-equiv=\\"X-UA-Compatible\\" content=\\"IE=edge\\">
    <meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1.0\\">
    <title>invalid html</title>
    <style>
        a{height:33px;color:#abc321;width:22px;}
    </style>
</head>
<body>
    
</body>
</html>"
`;

exports[`HTML pieces to be tested > valid.html 1`] = `
"<!DOCTYPE html>
<html lang=\\"en\\">
<head>
  <meta charset=\\"UTF-8\\">
  <meta http-equiv=\\"X-UA-Compatible\\" content=\\"IE=edge\\">
  <meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1.0\\">
  <title>valid html</title>
  <style>
    @import \\"x.css\\";
    @import \\"y.css\\";

    /**
    * Multi-line comment
    */

    :root {
      --brand-red: hsl(5deg 10% 40%);
    }

    .selector-1,
    .selector-2,
    .selector-3[type=\\"text\\"] {
      box-sizing: border-box;
      display: block;
      color: var(--brand-red);
      background: linear-gradient(#FFF, rgb(0 0 0 / 80%));
    }

    .selector-a,
    .selector-b:not(:first-child) {
      top: calc(100% - 2rem);
      padding: 10px !important;
    }

    .selector-x { width: 10%; }
    .selector-y { width: 20%; }
    .selector-z { width: 30%; }

    /* Single-line comment */

    @media (width >= 60em) {
      .selector {
        /* Flush to parent comment */
        transform: translate(1, 1) scale(3);
      }
    }

    @media (orientation: portrait), projection and (color) {
      .selector-i + .selector-ii {
        font-family: Helvetica, \\"Arial Black\\", sans-serif;
        background: hsl(20deg 25% 33%);
      }
    }

    /* Flush single line comment */
    @media
      screen and (min-resolution: 192dpi),
      screen and (min-resolution: 2dppx) {
      .selector {
        height: 10rem;
        margin: 10px;
        margin-bottom: 5px;
        background-image:
          repeating-linear-gradient(
            -45deg,
            transparent,
            #FFF 25px,
            rgb(255 255 255 / 100%) 50px
          );
        box-shadow:
          0 1px 1px #000,
          0 1px 0 #FFF,
          2px 2px 1px 1px #CCC inset;
        animation: 3s none fade-in;
      }

      /* Flush nested single line comment */
      .selector::after {
        content: \\"â†’\\";
        background-image: url(\\"x.svg\\");
      }
    }

    @keyframes fade-in {
      from { opacity: 0; }
      to { opacity: 1; }
    }

  </style>
</head>
<body>
  
</body>
</html>"
`;
